### PR TITLE
Linux scheduler testsuite

### PR DESCRIPTION
**Pull request by PraveenPenguin**
added linux-scheduler-testing as automative way test linux-scheduler

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
job log : [job.log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/916260/job.log.txt)

File: cpu/linsched.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            os.path.join(self.datadir, 'fix.patch'))
+        process.system(fix_patch, shell=True, ignore_status=True)
+
+        build.make(self.srcdir)
+
+    def test(self):
+
+        os.chdir(self.srcdir)
+
+        if process.system('./%s' % self.args, ignore_status=True, shell=True):
```
Comment (narasimhan-v) [inline cpu/linsched.py]: I think `process.run()` will be better here, since we need failure output as well.

File: cpu/linsched.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel(
+                    "Fail to install %s required for this test." % package)
+        git.get_repo('https://github.com/thejinxters/linux-scheduler-testing',
+                     destination_dir=self.srcdir)
+        os.chdir(self.srcdir)
+        fix_patch = 'patch -p1 < %s' % (
+            os.path.join(self.datadir, 'fix.patch'))
+        process.system(fix_patch, shell=True, ignore_status=True)
```
Comment (narasimhan-v) [inline cpu/linsched.py]: And I suggest `process.run()` here as well

File: cpu/linsched.py (hunk 1 above)
Comment (PraveenPenguin) [inline cpu/linsched.py]: @narasimhan-v i think should be fine as it is scheduler test so not sure here we can get any failure condition as after test run if command executed successfully user can verify out put of time take by scheduler

File: cpu/linsched.py (hunk 2 above)
Comment (PraveenPenguin) [inline cpu/linsched.py]: this can be done but not thinking about benefit of using `process.run` instead of `process.system` are you pointing we can check out put of patch applied ?

Comment (narasimhan-v) [inline cpu/linsched.py]: Yes, we need output of failure case

File: cpu/linsched.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        '''
+
+        # Check for basic utilities
+        smm = SoftwareManager()
+        deps = ['gcc', 'make', 'patch']
+        if distro.detect().name == "SuSE":
+            deps.append('git-core')
+        else:
+            deps.append('git')
+        self.args = self.params.get('args', default='pi 100')
```
Comment (apahim) [inline cpu/linsched.py]: this line should be after the `self.cancel()`

File: cpu/linsched.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        process.run(fix_patch, shell=True)
+
+        build.make(self.srcdir)
+
+    def test(self):
+
+        os.chdir(self.srcdir)
+
+        if process.system('./%s' % self.args, ignore_status=True, shell=True):
+            self.fail("test run fails of  %s" % self.args)
```
Comment (apahim) [inline cpu/linsched.py]: I'd suggest to rephrase this to something like:
```python
        cmd = './%s' % self.args
        if process.system(cmd, ignore_status=True, shell=True):
            self.fail('Command [%s] failed.' % cmd)
```

Comment (apahim) [inline cpu/linsched.py]: or just...
```python
        if process.system('./%s' % self.args, ignore_status=True, shell=True):
            self.fail('Test [%s] failed.' % self.args)
```

File: cpu/linsched.py (hunk 3 above)
Comment (PraveenPenguin) [inline cpu/linsched.py]: sure will do

File: cpu/linsched.py (hunk 4 above)
Comment (PraveenPenguin) [inline cpu/linsched.py]: make sense will change
